x-pack/plugins/discover_enhanced/public/actions/explore_data/explore_data_context_menu_action.ts

```diff
@@ -4,64 +4,27 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import type { Filter } from '@kbn/es-query';
+import { EmbeddableApiContext } from '@kbn/presentation-publishing';
 import { Action } from '@kbn/ui-actions-plugin/public';
-import { EmbeddableContext, EmbeddableInput, IEmbeddable } from '@kbn/embeddable-plugin/public';
-import type { Query, TimeRange } from '@kbn/es-query';
-import { DiscoverAppLocatorParams } from '@kbn/discover-plugin/common';
-import { KibanaLocation } from '@kbn/share-plugin/public';
-import * as shared from './shared';
 import { AbstractExploreDataAction } from './abstract_explore_data_action';
 
-interface EmbeddableQueryInput extends EmbeddableInput {
-  query?: Query;
-  filters?: Filter[];
-  timeRange?: TimeRange;
-}
-
-type EmbeddableQueryContext = EmbeddableContext<IEmbeddable<EmbeddableQueryInput>>;
-
 export const ACTION_EXPLORE_DATA = 'ACTION_EXPLORE_DATA';
 
 /**
  * This is "Explore underlying data" action which appears in the context
  * menu of a dashboard panel.
  */
 export class ExploreDataContextMenuAction
-  extends AbstractExploreDataAction<EmbeddableQueryContext>
-  implements Action<EmbeddableQueryContext>
+  extends AbstractExploreDataAction
+  implements Action<EmbeddableApiContext>
 {
   public readonly id = ACTION_EXPLORE_DATA;
 
   public readonly type = ACTION_EXPLORE_DATA;
 
   public readonly order = 200;
 
-  protected readonly getLocation = async (
-    context: EmbeddableQueryContext
-  ): Promise<KibanaLocation> => {
-    const { plugins } = this.params.start();
-    const { locator } = plugins.discover;
-
-    if (!locator) {
-      throw new Error('Discover URL locator not available.');
-    }
-
-    const { embeddable } = context;
-    const params: DiscoverAppLocatorParams = {};
-
-    if (embeddable) {
-      params.indexPatternId = shared.getIndexPatterns(embeddable)[0] || undefined;
-
-      const input = embeddable.getInput();
-
-      if (input.timeRange && !params.timeRange) params.timeRange = input.timeRange;
-      if (input.query) params.query = input.query;
-      if (input.filters) params.filters = [...input.filters, ...(params.filters || [])];
-    }
-
-    const location = await locator.getLocation(params);
-
-    return location;
-  };
+  public async isCompatible(context: EmbeddableApiContext): Promise<boolean> {
```

Comment:
Nit: I think this one could also be dropped.

Reply:
WOOPS, so sorry - doing things too quick. Done in c007b21d483cee974680a3bab383bb2c0b1892b0